action.yaml

```diff
@@ -70,14 +46,15 @@ runs:
       # Only run this if we don't already have a pre-commit on the PATH
       if: env.PYTHON_BIN == null && env.PRE_COMMIT_BIN == null && steps.pre-commit-config.outputs.exists != 'false'
       uses: actions/setup-python@v5
+    - name: Setup node
+      uses: actions/setup-node@v4
       with:
-        python-version: ${{ env.PYTHON_VERSION }}
-    - name: Install tools
-      # This will skip the (slow) pyenv install if we triggered the above
-      uses: open-turo/action-setup-tools@v1
+        # Default to node 20 but use the .node-version file if it exists in the consumer repo
+        node-version: "20"
+        node-version-file: .node-version
     - name: Commitlint
       # This version needs to be sync'd with the repo HEAD if a major version is cut
-      uses: open-turo/action-pre-commit/commitlint@v2
+      uses: open-turo/action-pre-commit/commitlint@f/make_action_pre_commit_faster
```

Comment:
There is not an easy way to pin the action major version here without checking out the repo, since GIthub doesn't allow expressions in the `uses`. The 2 options are:

* Embed the commitlint action here. I don't like it because it's too long
* Update this to v3 after the PR gets merged